src/pages/RoomMembersPage.tsx

```diff
@@ -173,6 +166,26 @@ function RoomMembersPage({report, policies}: RoomMembersPageProps) {
 
     const participants = useMemo(() => getParticipantsList(report, personalDetails, true), [report, personalDetails]);
 
+    useEffect(() => {
+        if (selectedMembers.length === 0 || !canSelectMultiple) {
+            return;
+        }
+
+        setSelectedMembers((prevSelectedMembers) => {
+            const updatedSelectedMembers = prevSelectedMembers.filter((accountID) => {
+                const isInParticipants = participants.includes(accountID);
+                const pendingChatMember = reportMetadata?.pendingChatMembers?.findLast((member) => member.accountID === accountID.toString());
+
+                const isPendingDelete = pendingChatMember?.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+
+                // Keep the member only if they're still in the room and not pending removal
+                return isInParticipants && !isPendingDelete;
+            });
+
+            return updatedSelectedMembers;
+        });
+    }, [participants, reportMetadata?.pendingChatMembers, canSelectMultiple, selectedMembers.length]);
```

Comment:
```suggestion
    }, [participants, reportMetadata?.pendingChatMembers, canSelectMultiple]);
```

We don't want to run this effect whenever a member is selected/deselected.